impl_core/src/main/java/io/opencensus/implcore/trace/export/SampledSpanStoreImpl.java

```diff
@@ -51,25 +51,44 @@
 
   private static final class Bucket {
 
-    private final EvictingQueue<SpanImpl> queue;
-    private long lastSampleNanoTime;
+    private final EvictingQueue<SpanImpl> sampledSpansQueue;
+    private final EvictingQueue<SpanImpl> notSampledSpansQueue;
+    private long lastSampledNanoTime;
+    private long lastNotSampledNanoTime;
 
     private Bucket(int numSamples) {
-      queue = EvictingQueue.create(numSamples);
+      sampledSpansQueue = EvictingQueue.create(numSamples);
+      notSampledSpansQueue = EvictingQueue.create(numSamples);
     }
 
     private void considerForSampling(SpanImpl span) {
       long spanEndNanoTime = span.getEndNanoTime();
-      // Need to compare by doing the subtraction all the time because in case of an overflow,
-      // this may never sample again (at least for the next ~200 years). No real chance to
-      // overflow two times because that means the process runs for ~200 years.
-      if (spanEndNanoTime - lastSampleNanoTime > TIME_BETWEEN_SAMPLES) {
-        queue.add(span);
-        lastSampleNanoTime = spanEndNanoTime;
+      if (span.getContext().getTraceOptions().isSampled()) {
+        // Need to compare by doing the subtraction all the time because in case of an overflow,
+        // this may never sample again (at least for the next ~200 years). No real chance to
+        // overflow two times because that means the process runs for ~200 years.
+        if (spanEndNanoTime - lastSampledNanoTime > TIME_BETWEEN_SAMPLES) {
```

Comment:
since this is copy/paster, the comment also is. One way to declutter that is to make a method that is make a method duration(long endNano, long beginNano) use it everywhere instead of direct subtract, if only to centralize the comment. That or delete the comment and wait 200years for someone to ask.